### PR TITLE
feat(fdc): Updated to allow for sdktype to be set (core vs gen)

### PR DESCRIPTION
**Pull request by maneesht**

### CODE REVIEW

File: packages/firebase_data_connect/firebase_data_connect/lib/src/common/common_library.dart [hunk 1]
```diff
@@ -12,6 +12,8 @@ import 'package:firebase_auth/firebase_auth.dart';
 part 'dataconnect_error.dart';
 part 'dataconnect_options.dart';
 
+enum ClientSDKType { core, generated }
```
Comment (aashishpatil-g) [inline packages/firebase_data_connect/firebase_data_connect/lib/src/common/common_library.dart:15]: This is named CallerSDKType in iOS and Kotlin. Can you rename this?

File: packages/firebase_data_connect/firebase_data_connect/lib/src/network/grpc_transport.dart [hunk 2]
```diff
@@ -59,9 +63,11 @@ class GRPCTransport implements DataConnectTransport {
     } catch (e) {
       log('Unable to get app check token: $e');
     }
+    String sdkTypeValue =
+        "flutter/${sdkType == ClientSDKType.core ? 'core' : 'gen'}";
```
Comment (aashishpatil-g) [inline packages/firebase_data_connect/firebase_data_connect/lib/src/network/grpc_transport.dart:67]: I think the value here would be flutter/core and flutter/gen. 

This blank if sdk is core. 

If gen the format is lang/gen => dart/gen

Comment (aashishpatil-g) [reply packages/firebase_data_connect/firebase_data_connect/lib/src/network/grpc_transport.dart:67]: Pinged you the internal chat thread on this

**Review by aashishpatil-g — CHANGES_REQUESTED** [review]

**Review by aashishpatil-g — APPROVED** [review]
Looks good pending the requested changes.

File: packages/firebase_data_connect/firebase_data_connect/lib/src/network/grpc_transport.dart [hunk 3]
```diff
@@ -59,9 +63,11 @@ class GRPCTransport implements DataConnectTransport {
...
     Map<String, String> metadata = {
       'x-goog-request-params': 'location=${options.location}&frontend=data',
-      'x-goog-api-client': 'gl-dart/flutter fire/$packageVersion'
+      'x-goog-api-client': 'gl-dart/flutter fire/$packageVersion $sdkTypeValue'
```
Comment (aashishpatil-g) [inline packages/firebase_data_connect/firebase_data_connect/lib/src/network/grpc_transport.dart]: For our other platforms we use `gl-lang/`. Is the /flutter after gl-dart/ required? 

Example: https://github.com/firebase/firebase-js-sdk/pull/8480/files#diff-e6ca47cb91a2b2e0305e689ab4bad8faceff2a9370065604c953a838e3d69fabR27

File: packages/firebase_data_connect/firebase_data_connect/lib/src/common/common_library.dart [hunk 4]
```diff
@@ -15,7 +15,7 @@ part 'dataconnect_options.dart';
 enum CallerSDKType { core, generated }
 
 String getGoogApiVal(CallerSDKType sdkType, String packageVersion) {
-  String apiClientValue = 'gl-dart/flutter fire/$packageVersion';
+  String apiClientValue = 'gl-dart/$packageVersion fire/$packageVersion';
```
Comment (aashishpatil-g) [inline packages/firebase_data_connect/firebase_data_connect/lib/src/common/common_library.dart:18]: the one after gl-dart/ is supposed to be dart language version. So it should be gl-dart/DART_VERSION. 

If we can't get dart version lets leave it blank like we do with gl-js/

Comment (aashishpatil-g) [reply packages/firebase_data_connect/firebase_data_connect/lib/src/common/common_library.dart:18]: https://github.com/firebase/data-connect-ios-sdk/blob/main/Sources/Internal/GrpcClient.swift#L84

Comment (maneesht) [reply packages/firebase_data_connect/firebase_data_connect/lib/src/common/common_library.dart:18]: Looks like for flutter we just repeat the packageVersion: https://github.com/firebase/flutterfire/blob/ff1e5f672cee29731dc4d21251611030add9e605/packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_model.dart#L97

**Review by cynthiajoan — APPROVED** [review]

File: packages/firebase_data_connect/firebase_data_connect/lib/src/common/common_library.dart (hunk 4 above)
Comment (aashishpatil-g) [reply packages/firebase_data_connect/firebase_data_connect/lib/src/common/common_library.dart:18]: Sgtm then.

**Review by russellwheatley — APPROVED** [review]

**Pull request merged**